### DISCUSSION TITLE
hjkl movements for specific buffers like vterm,ansi-term,shell

### DISCUSSION THREAD

**Discussion by underside** [meow-edit/meow#519] [General]
When I have to use Evil, vterm buffers feels like ordinary buffers.

IN NORMAL or VISUAL modes: move around with hjkl, select part of terminal output, copy it to  another buffer e.t.c 
IN INSERT mode: input terminal commands

It's very critical feature for me since I use vterm massively and regularly copy/paste output. 

Can it be achieved in meow?

**Comment by okamsn** [comment]
I don't use Vterm, but this seems to do want you describe:

```emacs-lisp
(with-eval-after-load 'meow
  (push '(vterm-mode . insert) meow-mode-state-list)
  (add-hook 'vterm-mode-hook
            (lambda ()
              (add-hook 'meow-insert-enter-hook
                        (lambda () (vterm-copy-mode -1))
                        nil t)
              (add-hook 'meow-insert-exit-hook
                        (lambda () (vterm-copy-mode 1))
                        nil t))))
```
```

**Reply by underside** [reply]
OMG, it's absolutely beautiful!
Thank you!
